src/io/syncfile.c

```diff
@@ -37,6 +37,9 @@ typedef struct {
     /* Is it known to be writable? */
     short known_writable;
 
+    /* Did the open mode specify O_RDONLY, O_WRONLY, or O_RDWR? (ignores other mode settings) */
+    short open_mode;
```

Comment:
If you just move open_mode to right after eof_reported, the whole struct shrinks by 8 bytes.

Reply:
Done!